guardian/src/main/java/org/rfcx/guardian/guardian/activity/MainActivity.kt

```diff
@@ -96,51 +84,44 @@ class MainActivity : AppCompatActivity() {
         }
 
         registerButton.setOnClickListener {
-            if (isNetworkAvailable(this)) {
+            if (GuardianUtils.isNetworkAvailable(this)) {
                 if (this.getTokenID() != null) {
-                    if (!isGuidExisted()) {
+                    if (!GuardianUtils.isGuidExisted(this)) {
                         setVisibilityBeforeRegister()
                         val guid = app.rfcxDeviceGuid.deviceGuid
                         val token = app.rfcxDeviceGuid.deviceToken
-                        Log.d("GuidInfo", app.rfcxDeviceGuid.deviceGuid)
-                        Log.d("GuidInfo", app.rfcxDeviceGuid.deviceToken)
                         RegisterApi.registerGuardian(
                             applicationContext,
                             RegisterRequest(guid, token),
                             object :
                                 RegisterCallback {
                                 override fun onSuccess() {
-                                    GuardianCheckApi.exists(applicationContext, guid, object:
+                                    GuardianCheckApi.exists(applicationContext, guid, object :
                                         CheckGuardianCallback {
                                         override fun onSuccess() {
-                                            createRegisterFile(app)
-                                            setUIByRecordingState(app)
-                                            setUIByGuidState(app)
                                             setVisibilityRegisterSuccess()
-                                            deviceIdText.text = readRegisterFile()
+                                            GuardianUtils.createRegisterFile(baseContext)
                                             app.startAllServices()
+                                            setUIByRecordingState()
+                                            setUIByGuidState()
                                             setUIFromBtnClicked("start")
-                                            getCheckinInformation(app)
+                                            getCheckinInformation()
+                                            deviceIdText.text = GuardianUtils.readRegisterFile(baseContext)
```

Comment:
This code is nested to 8+ indents... A better design would be to separate our your nested callbacks.